### DISCUSSION TITLE
Install as an unRAID App

### DISCUSSION THREAD

**Discussion by alex3305** [louislam/dockge#238] [General]
As I was trying out Dockge (looks great so far!), I noticed that there wasn't a template for an unRAID deployment. So I decided to [create my own](https://github.com/alex3305/unraid-docker-templates) for use in the unRAID Community Applications. So searching for dockge will let you easily install it 😄!

![image](https://github.com/louislam/dockge/assets/5155694/0b850878-5203-475e-9aa0-5863e14a0175)

If anyone has any tips or suggestion I'm happy to accept those. Thanks for all the great work!

**Comment by louislam** [comment]
Awesome! Thank you.

**Comment by Zaid-maker** [comment]
Amazing Although i dont use unRAID but this sounds good that our community is great ❤️

**Comment by Magneticdud** [comment]
Hi I installed your dockge app in my unRAID server but containers managed by dockge don't auto start after a reboot.

In docker compose they're set as "unless stopped" and in unRAID docker page they have the toggle on the auto start.

Is this normal or I did some mistake? I much prefer dockge because it's so much faster

**Reply by jammsen** [reply]
Yes thats right, Unraid uses its own "Docker-Management" sadly.

**Reply by louislam** [reply]
I was experienced this too previously, but it is working now for me suddenly.

It maybe fixed by the latest version of Unraid (6.12.13), or I installed the Docker Compose Manager plugin by dcflachs.

I also changed my Docker engine from vdisk to directory.

<img width="1059" alt="1235592024_Bildschirmfoto2022-10-25um21_39_22 png 9961b2a821de1f457cef326966d95093" src="https://github.com/user-attachments/assets/d3283843-2b58-460c-9386-9f2f001e9227">

**Reply by alex3305** [reply]
For me it seems that sometimes it works, sometimes it doesn't. I suspect it's dependent on the network setup. That also seems a little bit on the flakey side 😛.

**Reply by jammsen** [reply]
> I was experienced this too previously, but it is working now for me suddenly.
> 
> It maybe fixed by the latest version of Unraid (6.12.13), or I installed the Docker Compose Manager plugin by dcflachs.
> 
> I also changed my Docker engine from vdisk to directory.
> 
> <img alt="1235592024_Bildschirmfoto2022-10-25um21_39_22 png 9961b2a821de1f457cef326966d95093" width="1059" src="https://private-user-images.githubusercontent.com/1336778/374170049-d3283843-2b58-460c-9386-9f2f001e9227.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgzMjcwMDcsIm5iZiI6MTcyODMyNjcwNywicGF0aCI6Ii8xMzM2Nzc4LzM3NDE3MDA0OS1kMzI4Mzg0My0yYjU4LTQ2MGMtOTM4Ni05ZjJmMDAxZTkyMjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDdUMTg0NTA3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2U1N2E2MTYyODVjYTc0OWQ5YTdiZGE1ZGFiNGZkMjdlZDc4MTI4OTU3YWE2ZDUyYWRkNTM0NTYyZDhmNjgwNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4nEOu3oOyxtei1up3exovtgktd6Qejv9ZBO5tRHzUDg">

If you do App-Provision via the community app-store or via "Docker Compose" it works good for me. I have "User Scripts" from Squid installed and at the Event of "At First Array Start Only" im using this code:
```
#!/bin/bash
cd /mnt/user/appdata/autostart
./get-docker-compose.sh
```

get.docker-compose.sh
```shell
#!/bin/bash
curl -SL https://github.com/docker/compose/releases/download/v2.24.7/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
chmod +x /usr/local/bin/docker-compose
```

This works just fine for me, here and there i upgrade to a never binary.

**Reply by dada051** [reply]
Why get.docker-compose.sh in User scripts instead of the docker-compose plugin in the store?

**Reply by alex3305** [reply]
@dada051 The Docker Compose plugin in the store is IMHO unnecessarily bloated. For my install I converted it to an Ansible role with the following step:

```yaml
- name: Install Docker Compose
  ansible.builtin.get_url:
    url: "https://github.com/docker/compose/releases/download/v2.34.0/docker-compose-linux-x86-64"
    dest: "/usr/local/lib/docker/cli-plugins/docker-compose"
    owner: "root"
    group: "docker"
    mode: "0755"
    checksum: "sha256:https://github.com/docker/compose/releases/download/v2.34.0/docker-compose-linux-x86-64.sha256"
```

My local Ansible role is a little more complicated, because I also consider other architectures (ie. Pi's) and have templated the version so I can bump it with Renovate. But it's pretty trivial to do something like this in a bash script and set correct permissions, which would be something like this:

```bash
curl -sSLo /usr/local/lib/docker/cli-plugins/docker-compose https://github.com/docker/compose/releases/download/v2.34.0/docker-compose-linux-x86-64
chown root:docker /usr/local/lib/docker/cli-plugins/docker-compose 
chmod 0755 /usr/local/lib/docker/cli-plugins/docker-compose 
```

If you compare that to the [Compose Plugin install script](https://github.com/dcflachs/compose_plugin/blob/main/source/pkg_build.sh) that's way more complicated. And if you are not going to use the plugin frontend functionality anyways I don't see a reason to not use a simple bash script. There is nothing wrong with Compose Plugin btw. The plugin is excellent work but I understand other solutions.

**Comment by dada051** [comment]
I'm maybe wrong, but it looks like that dockge on unRAID 7 with works OOB.
